### PR TITLE
reputation: Add configuration parameters

### PR DESCRIPTION
**Pull request by ifraixedes**
__What__:

Add the configuration parameters which will be used by the algorithm
which will calculate the storage node reputation.

__Why:__

Because the algorithm to calculate the SNs reputation require them as stated in draft [Reputation & Node Selection design doc](https://github.com/storj/storj/pull/2135)

__Other concerns:__

The reputation calculation is based on audit and uptime check
results hence some configuration parameters are in pkg/audit, others in
pkg/discovery and others in the satellite which will combine the both
reputation results to obtain the storage node reputation for repair and
uplink.

__NOTE__: only a few default configuration values are specified in the document, the rest of them I don't know which are the good default values, so I set them to `1.0` and waiting in the review to know which are the correct default values.

Thanks for submitting a PR!

## Code Review Checklist (to be filled out by reviewer)
 - [ ] Does the PR describe what changes are being made?
 - [ ] Does the PR describe why the changes are being made?
 - [ ] Does the code follow [our style guide](https://github.com/storj/docs/blob/master/code/Style.md)?
 - [ ] Does the code follow [our testing guide](https://github.com/storj/docs/blob/master/code/Testing.md)?
 - [ ] Could the PR be broken into smaller PRs?
 - [ ] Does the new code have enough tests? (*every* PR should have tests or justification otherwise. Bug-fix PRs especially)
 - [ ] Does the new code have enough documentation that answers "how do I use it?" and "what does it do?"? (both source documentation and [higher level](https://github.com/storj/docs), diagrams?)
 - [ ] Does any documentation need updating?

### CODE REVIEW

**Comment by ifraixedes** [comment]
The CI fails due to satellite configuration has changed and the config lock file isn't updated so far, because the review will confirm if this configuration parameters are correct and are in the right place.

File: pkg/audit/service.go [hunk 1]
```diff
@@ -30,6 +30,11 @@ type Config struct {
 	MinBytesPerSecond  memory.Size   `help:"the minimum acceptable bytes that storage nodes can transfer per second to the satellite" default:"128B"`
 	MinDownloadTimeout time.Duration `help:"the minimum duration for downloading a share from storage nodes before timing out" default:"5s"`
 	MaxReverifyCount   int           `help:"limit above which we consider an audit is failed" default:"3"`
+
+	ReputationAlpha0 float64 `help:"the initial shape 'alpha' used to calculate audit SNs reputation" default:"1.0"`
```
Comment (navillasa) [inline pkg/audit/service.go]: why Alpha0 and Beta0 instead of just Alpha and Beta?

Comment (wthorp) [reply pkg/audit/service.go]: The zero is meant to show that this is the initial value, rather than the value at the Nth measurement.

**Comment by navillasa** [comment]
since these reputation ratios will be used for node selection, I'm thinking they could potentially be better located in the `pkg/overlay/config.go` file.

**Comment by ifraixedes** [comment]
> since these reputation ratios will be used for node selection, I'm thinking they could potentially be better located in the pkg/overlay/config.go file.

At the end they need to be used for calculating the _uptime_ and _audit_ reputation which will be used for calculating the total _repair_ and _uplink_ reputation, these 2 last ones will be the one used for selecting and disqualify nodes.

If _pkg/overlay_ package is the package which will do those calculations, then yes, they should be there.

I cannot say because I'm not still very clear in the how the sources are split across packages in the project, so I cannot have an opinion on why not to move them to the _pkg/overlay_, but I would like that another reviewer have the same opinion before doing the change.

**Comment by ifraixedes** [comment]
As commented in other means, the configuration parameters have been moved to _pkg/overlay_ as @navillasa suggested.

I'm going to warn our Ops team about the new configuration parameters.

**Pull request merged**